guppylang/compiler/cfg_compiler.py

```diff
@@ -149,20 +153,21 @@ def choose_vars_for_tuple_sum(
     return conditional.add_out_port(SumType([row_to_type(row) for row in tys]))
 
 
-def compare_var(x: Variable, y: Variable) -> int:
+def compare_var(p1: Place, p2: Place) -> int:
     """Defines a `<` order on variables.
 
     We use this to determine in which order variables are outputted from basic blocks.
     We need to output linear variables at the end, so we do a lexicographic ordering of
     linearity and name. The only exception are return vars which must be outputted in
     order.
     """
-    if is_return_var(x.name) and is_return_var(y.name):
-        return -1 if x.name < y.name else 1
-    return -1 if (x.ty.linear, x.name) < (y.ty.linear, y.name) else 1
+    x, y = str(p1), str(p2)
+    if is_return_var(x) and is_return_var(y):
+        return -1 if x < y else 1
+    return -1 if (p1.ty.linear, x) < (p2.ty.linear, y) else 1
```

Comment:
Indeed, this `%ret` business is quite hacky. This currently works since return vars are only compared with each other, hower this changes with `@inout`.

This special handling of return vars is removed in #340, so I think it's not really worth putting effort into a nice solution here though